intermediate_notebooks/examples/custreamz/parsing_haproxy_logs.ipynb

```diff
@@ -0,0 +1,427 @@
+    "    clean_df.drop_column(17)\n",
+    "    del q_df\n",
+    "    \n",
+    "    clean_df['time_backend_response'] = clean_df['time_backend_response'].astype('int')\n",
+    "    clean_df['http_status_code'] = clean_df['http_status_code'].astype('int')\n",
+    "    \n",
+    "    # Return the cuDF dataframe so that other operations in the stream can use it\n",
+    "    return clean_df\n",
+    "    \n",
+    "# We now use Streamz to create a Stream from Kafka by polling the topic every 1s. If you changed Dask=True, please ensure you have a Dask cluster up and running\n",
+    "source = Stream.from_kafka_batched(topic, kafka_conf, npartitions=1, poll_interval='1s', asynchronous=True, dask=False)\n",
+    "\n",
+    "# Maintain a list of DataFrames\n",
+    "gdfs = []\n",
+    "\n",
+    "# When started this returns to us a RAPIDS cuDF\n",
+    "#sdf = source.map(haproxy_lines_to_cudf).sink(lambda sdf: gdfs.append(sdf))\n",
+    "sdf = source.map(haproxy_lines_to_cudf)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above snippet defines a function for parsing a batch of HAProxy logs into a cuDF and also defining streamz code that will be used to invoke that function. Keep in mind the Stream is not running until we later invoke \"start\" so right now we are still just building up our configuration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Counts the number of 403 from the \"http_status_code\"\n",
+    "def count_auth_errors(gdf):\n",
+    "    auth_errors = gdf[gdf['http_status_code'] == 403].http_status_code.count()\n",
+    "    return auth_errors\n",
+    "\n",
+    "# Count the number of 200 codes from \"http_status_code\". Of course it would be better to make a common function for this but being verbose for demonstration purposes.\n",
+    "def count_ok_codes(gdf):\n",
+    "    ok_codes = gdf[gdf['http_status_code'] == 200].http_status_code.count()\n",
+    "    return ok_codes\n",
+    "\n",
+    "# Here we demonstrate splitting from the parent \"sdf\" stream source and branch out to different streams.\n",
+    "auth_codes = sdf.map(count_auth_errors).sink(lambda err_count: print(\"{:,} - 403 codes were encountered.\".format(err_count)))\n",
+    "ok_codes = sdf.map(count_ok_codes).sink(lambda ok_count: print(\"{:,} - 200 codes were encountered.\".format(ok_count)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Calculating Average Backend Response Time\n",
+    "\n",
+    "Our code above has already taken care of creating a \"Streaming DataFrame\" using RAPIDS cuDF for us. So no need to do that again. What we can do is reuse that object and inquire different questions on it with another stream without having to pull all of the messages from Kafka again (slowest part of pipeline). Lets take a look at that."
```

Comment:
It would be helpful to link to the Streamz doc page about what a Streaming DataFrame is